src/cpp/core/system/PosixSystem.cpp

```diff
@@ -2426,11 +2429,64 @@ Error runProcess(const std::string& path,
    return error;
 }
 
-Error getChildProcesses(std::vector<ProcessInfo> *pOutProcesses, bool populateUsername)
+Error getChildProcesses(
+      std::vector<ProcessInfo>* pOutProcesses,
+      bool populateUsername)
 {
    return getChildProcesses(::getpid(), pOutProcesses, populateUsername);
 }
 
+#ifdef __APPLE__
+
+Error getChildProcesses(
+      pid_t pid,
+      std::vector<ProcessInfo>* pOutProcesses,
+      bool populateUsername)
+{
+   if (!pOutProcesses)
+      return systemError(EINVAL, ERROR_LOCATION);
+
+   // get child processes
+   const std::size_t N = 1024;
+   std::vector<pid_t> pids;
+   pids.resize(N);
```

Comment:
In the "makes no real difference" category I _think_ you can specify the size in the constructor instead of a separate resize, i.e. `std::vector<pid_t> pids(N)`. Didn't check.

Reply:
You are quite right -- that's probably a better approach.